Update TS docker build information (#2314)
temporalio/documentation: docs-src/typescript/how-to-run-a-worker-on-docker.md

  - typescript
---

:::note

> Before proceeding, we recommend referring to our [production sample](https://github.com/temporalio/samples-typescript/tree/main/production),

Reviewer: I'd reverse that sentence. Our recommendation is really that they prepare the workflow bundle ahead-of-time, not that they look at the production sample.

```suggestion
To improve worker startup time, we recommend that workflow bundles be prepared ahead-of-time. See our [production sample](https://github.com/temporalio/samples-typescript/tree/main/production) for details.
```

Author: Changed to more or less what you're suggesting.

which shows how to save the compiled workflow bundle to a file and avoid compilation overhead on worker startup.

:::

Workers based on the TypeScript SDK can be deployed and run as Docker containers.

We recommend an LTS Node.js release such as 18 or 20.
Both `amd64` and `arm64` architectures are supported.
A glibc-based image is required; musl-based images are _not_ supported (see below).

The easiest way to deploy a TypeScript SDK Worker on Docker is to start with the `node:20-bullseye` image.
For example:

```dockerfile
FROM node:20-bullseye

# For better cache utilization, copy package.json and lock file first and install the dependencies before copying the
# rest of the application and building.
COPY . /app
WORKDIR /app

> # Alternatively, run npm ci, which installs only dependencies specified in the lock file and is generally faster.

Reviewer: I think at this point, it should be reasonable to recommend this instead:

```
npm ci --omit dev
```
`--omit` was introduced in NPM 8, which itself got introduced in 16.11.0.

Author: `--omit dev` would require a separate build step, I'd avoid that for now.

CMD ["build/worker.js"]
```

> For smaller images and/or more secure deployments, it is also possible to use `-slim` Docker image variants (like `node:20-bullseye-slim`) or `distroless/nodejs` Docker images (like `gcr.io/distroless/nodejs:20`) with the following caveats.

Reviewer: Looks like the distroless project changed their URL structures starting with Node 20.

```suggestion
For smaller images and/or more secure deployments, it is also possible to use `-slim` Docker image variants (like `node:20-bullseye-slim`) or `distroless/nodejs` Docker images (like `gcr.io/distroless/nodejs20-debian11`) with the following caveats.
```

# -- RESULTING IMAGE --

> FROM gcr.io/distroless/nodejs:20

Reviewer: ```suggestion
FROM gcr.io/distroless/nodejs20-debian11
```
